assets/js/custom/AdminMail.js

```diff
@@ -2,22 +2,55 @@
 
 // eslint-disable-next-line no-unused-vars
 function AdminMail () {
-  $('.btn').click(function () {
-    const username = $('#username').val()
-    const subject = $('#subject').val()
-    const message = $('#content').val()
-    const resultBox = $('.resultBox')
+  const sendBtn = document.querySelector('.btn-send')
+  const previewBtn = document.querySelector('.btn-preview')
+  const usernameInput = document.querySelector('#username')
+  const subjectInput = document.querySelector('#subject')
+  const titelInput = document.querySelector('#titel')
+  const messageInput = document.querySelector('#content')
+  const templateSelect = document.querySelector('#template-select')
+  const resultBox = document.querySelector('.resultBox')
 
-    // calls _/Controller/Admin/Tools/SendMailToUserController::sendAction
-    $.get('send', { username, subject, message }, function (data) {
-      if (data && data.length >= 2 && data.substring(0, 2) === 'OK') {
-        resultBox.removeClass('error')
-        resultBox.addClass('success')
-      } else {
-        resultBox.removeClass('success')
-        resultBox.addClass('error')
-      }
-      resultBox.html(data)
-    })
-  })
+  function sendMail () {
+    const username = usernameInput.value
+    const subject = subjectInput.value
+    const titel = titelInput.value
+    const message = messageInput.value
+    const url = `send?username=${username}&subject=${subject}&titel=${titel}&message=${message}`
```

Comment:
This will break on specific inputs. For example, let's consider you have the following example:

username = "user1"
subject = "Password Reset"
title = "...
....

Then you will get the following URL:

URL = "send?username=user1&subject=Password Reset&title=...

Since the parameters are not url-encoded, characters such as the space character will break your URL resulting into just fetching "send?username=user1&subject=Password"

Reply:
window.open() encodes the url parameter by itself

Reply:
Is this still an issue? I didnt see any problem during testing, is it resolved @dmetzner ?